test/dist_autograd_test.py

```diff
@@ -689,6 +689,22 @@ def test_backward_node_failure(self):
                 # Kill all other nodes.
                 sys.exit(0)
 
+    @dist_init(clean_shutdown=False)
+    def test_context_cleanup_timeout(self):
+        global known_context_ids
+        dst_ranks = {rank for rank in range(self.world_size) if rank != self.rank}
+        with dist_autograd.context() as context_id:
+            t1 = torch.rand((3, 3), requires_grad=True)
+            t2 = torch.rand((3, 3), requires_grad=True)
+
+            time.sleep(300);
+            if self.rank == 2:
+                pass
+            else:
+                with self.assertRaises(RuntimeError):
```

Comment:
might be better to write this as
`if self.rank != 2: with self.assertRaises(...)` to reduce some verbosity?